ruleset/rules/0017-wazuh-api_rules.xml

```diff
@@ -1,178 +1,196 @@
+    <description>Wazuh API: $(endpoint) response code returned error.</description>
+    <group>gpg13_4.3</group>
+  </rule>
+
+  <rule id="410" level="4">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">400</field>
+    <description>Wazuh API: $(endpoint) Bad request.</description>
+  </rule>
+
+  <rule id="411" level="8">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">401</field>
+    <description>Wazuh API: $(endpoint) Unauthorized.</description>
+    <group>gpg13_4.3,tsc_cc6.8,hipaa_164.312.e.1</group>
+  </rule>
+
+  <rule id="412" level="7">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">403</field>
+    <description>Wazuh API: $(endpoint) Permission denied.</description>
+    <group>gpg13_4.3,hipaa_164.312.e.1,nist_800_53_AC.7,pci_dss_7.1,pci_dss_10.2.4,tsc_cc6.8,</group>
+  </rule>
+
+  <rule id="413" level="4">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">404</field>
+    <description>Wazuh API: $(endpoint) Resource not found.</description>
+  </rule>
+
+  <rule id="414" level="4">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">405</field>
+    <description>Wazuh API: $(endpoint) Invalid HTTP method.</description>
+  </rule>
+
+  <rule id="415" level="4">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">406</field>
+    <description>Wazuh API: $(endpoint) Invalid content-type.</description>
+  </rule>
+
+  <rule id="416" level="4">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">413</field>
+    <description>Wazuh API: $(endpoint) Maximum request body size exceeded.</description>
+  </rule>
+
+  <rule id="417" level="7">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">429</field>
+    <description>Wazuh API: $(endpoint) Max number of requests per minute reached.</description>
+    <group>gpg13_4.3,tsc_cc6.8,hipaa_164.312.e.1,nist_800_53_AC.7,pci_dss_7.1,pci_dss_8.1.6,pci_dss_10.2.4,</group>
+  </rule>
+
+  <rule id="418" level="4">
+    <if_sid>405</if_sid>
+    <field name="http_status_code">500</field>
+    <description>Wazuh API: $(endpoint) Internal error.</description>
+  </rule>
+
+  <rule id="406" level="4">
+    <if_sid>400</if_sid>
+    <field name="method">GET</field>
+    <description>Wazuh API: $(endpoint) Success.</description>
+  </rule>
+
+  <rule id="407" level="5">
+    <if_sid>400</if_sid>
+    <field name="method">PUT</field>
+    <description>Wazuh API:$(endpoint) Success.</description>
+  </rule>
+
+  <rule id="408" level="7">
+    <if_sid>400</if_sid>
+    <field name="method">DELETE</field>
+    <description>Wazuh API: $(endpoint) Success.</description>
+    <group>hipaa_164.312.a.2.I,hipaa_164.312.c.1,pci_dss_8.1.2,</group>
+  </rule>
+
+  <rule id="409" level="5">
+    <if_sid>400</if_sid>
+    <field name="method">POST</field>
+    <description>Wazuh API: $(endpoint) Success.</description>
+    <group>hipaa_164.312.c.1</group>
+  </rule>
```

Comment:
Rules are not correctly ordered by their ID

Reply:
And groups inside "group" labels are also not correctly ordered

Reply:
done